From ad6c9b287742e49b9647ecfc31191fe5493dc7d3 Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Sun, 7 Aug 2016 13:48:08 +0300
Subject: [PATCH 0392/1345] debugfs: Add CPU Aux Ctrl system register to regs
 dump

commit  12e89f5e1e1711daa3502b276770337aed64eec9 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

In addition, expand printputs to 64-bits as this register is a
64-bit regier.

Change-Id: Ie5742db3a8b6df06680656573f3884ec97670d49
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/31769
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/misc/mvebu-a8k-debugfs.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/mvebu-a8k-debugfs.c b/drivers/misc/mvebu-a8k-debugfs.c
index dea90da..3a95423 100644
--- a/drivers/misc/mvebu-a8k-debugfs.c
+++ b/drivers/misc/mvebu-a8k-debugfs.c
@@ -63,6 +63,7 @@ struct sysreg_entry {
 DECLARE_RD_WR_FUNC(s3_1_c11_c0_2);
 DECLARE_RD_WR_FUNC(s3_1_c11_c0_3);
 DECLARE_RD_WR_FUNC(s3_1_c15_c2_1);
+DECLARE_RD_WR_FUNC(s3_1_c15_c2_0);
 
 
 #define SYSREG_ENTRY(code, _desc)		\
@@ -81,6 +82,7 @@ struct sysreg_entry {
 	SYSREG_ENTRY(s3_1_c11_c0_2, "L2 Ctrl"),
 	SYSREG_ENTRY(s3_1_c11_c0_3, "L2 Ext Ctrl"),
 	SYSREG_ENTRY(s3_1_c15_c2_1, "CPU Ext Ctrl"),
+	SYSREG_ENTRY(s3_1_c15_c2_0, "CPU Aux Ctrl"),
 };
 
 
@@ -93,7 +95,7 @@ static int mvebu_ap806_sysregs_debug_show(struct seq_file *seq, void *v)
 	for (i = 0; i < ARRAY_SIZE(sysregs_list); i++) {
 		seq_printf(seq, "%s\t\t- ", sysregs_list[i].name);
 		val = sysregs_list[i].read_func();
-		seq_printf(seq, "0x%08llx", val);
+		seq_printf(seq, "0x%016llx", val);
 		seq_printf(seq, " (%s).\n", sysregs_list[i].desc);
 	}
 
@@ -152,7 +154,7 @@ static ssize_t mvebu_ap806_sysregs_debug_write(struct file *f, const char __user
 
 		sysregs_list[i].write_func(val64);
 
-		pr_info("New value %s - 0x%08llx.\n", sysregs_list[i].name,
+		pr_info("New value %s - 0x%016llx.\n", sysregs_list[i].name,
 				sysregs_list[i].read_func());
 	} else {
 		pr_err("Bad system register name.\n");
-- 
1.7.9.5

